compiler/src/dotty/tools/dotc/typer/Namer.scala

```diff
@@ -1851,7 +1851,11 @@ class Namer { typer: Typer =>
         }
         val defaultAlts = meth.altsWith(_.hasDefaultParams)
         if (defaultAlts.length == 1)
-          paramProto(defaultAlts.head.info.widen.paramInfoss, idx)
+          meth.info match
+            case info: PolyType =>
+              paramProto(defaultAlts.head.info.widen.paramInfoss, idx).substParams(info, paramss.head.map(_.typeRef))
```

Comment:
This substitution should only be done if expectedDefaultArgType == originalTp, since we don't want to compute expectedDefaultArgType multiple times it might need to become a lazy val, or the code needs to be refactored in another way.